src/changefile/promptForChange.ts

```diff
@@ -93,9 +93,15 @@ export async function promptForChange(options: BeachballOptions) {
         return;
       }
 
-      // fallback to the options.type if type is absent in the user input
-      if (!response.type && options.type) {
-        response = { ...response, type: options.type };
+      // if type is absent in the user input, there are two possiblities for
+      // proceeding next:
+      // 1) if options.type is defined, use that
+      // 2) otherwise, we hit the edge case when options.type is undefined
+      //    and there was only one possible ChangeType to display, 'none'
```

Comment:
Thanks for figuring this out! A clarifying question: I thought in the specific case you encountered, there should have been two possible change types (none and prerelease)? Or am I misunderstanding?

Reply:
Hmm, interesting! There actually was only `'none'` as an option available. If there were two options to choose from, the bug would not surface. Lemme quickly have a look why `'prelease'` was not available as an option in this situation.

Reply:
Hmm, ok, so `semver.prelease` returns `null` since the package version is just a triple, and it expects a quadruple, e.g., `0.1.0-pre`. We can continue the conversation going in the PR proper https://github.com/microsoft/fluentui/pull/16458.

Reply:
Hmm I think you've found an interesting edge case which we might want @kenotron's input on how to handle.

In Fluent UI right now we have a repo-wide config with `disallowedChangeTypes: ['major', 'minor', 'patch']`. This implicitly assumes that all published packages have a prerelease semver. For the package that had changes, that wasn't the case, which is why beachball only offered `none` as an option.

Your change here is one possible way to better handle that case, but I wonder if an even better way would be to have beachball show an error? (suggesting to either add a prerelease to the package's version, or add a package-specific override to allow more change types)

Reply:
If you look closely, this is not an erroneous condition, and shouldn't be treated as such. First of all, in [promptForChange.ts#L49](https://github.com/microsoft/beachball/blob/f7f6170068a9e8548c9bff62f05c316357b65918/src/changefile/promptForChange.ts#L49) we ensure that we have at least one valid option available. Then, in [promptForChange.ts#L68](https://github.com/microsoft/beachball/blob/f7f6170068a9e8548c9bff62f05c316357b65918/src/changefile/promptForChange.ts#L68), we ensure that we only show the `ChangeType` prompt if we have at least 2 valid choices, meaning, the case when both `!options.type` evaluates to `true`, and having one option available, should either be shown to the end-user, or should override `options.type` with that option. This is in fact already enforced for situations when `!options.type` evaluates to `true`, and there is only one valid `ChangeType` option available, and `!option.message` also evaluates to `true`. In this case, we completely bypass the prompts in [promptForChange.ts#88](https://github.com/microsoft/beachball/blob/f7f6170068a9e8548c9bff62f05c316357b65918/src/changefile/promptForChange.ts#L88), and set the `response` to the defaults in [promptForChange.ts#L83](https://github.com/microsoft/beachball/blob/f7f6170068a9e8548c9bff62f05c316357b65918/src/changefile/promptForChange.ts#L83) where `type := 'none'`.